qiskit/qpy/interface.py

```diff
@@ -69,6 +69,7 @@
 """
     + "$"
 )
+VERSION_PATTERN_REGEX = re.compile(VERSION_PATTERN)
```

Comment:
Oh, the flags probably need moving out of the `search` calls and into the `compile`.

Reply:
Oh good call, completely missed that. Let me fix that right now

Reply:
Fixed in: https://github.com/Qiskit/qiskit-terra/pull/8235/commits/2760a38968b2ded1bd4cc2bf25665ec1e0a337df